cpp/src/arrow/util/decimal_benchmark.cc

```diff
@@ -26,14 +26,36 @@
 namespace arrow {
 namespace Decimal {
 
-static void FromString(benchmark::State& state) {  // NOLINT non-const reference
-  std::vector<std::string> values = {"0",
-                                     "1.23",
-                                     "12.345e6",
-                                     "-12.345e-6",
-                                     "123456789.123456789",
-                                     "1231234567890.451234567890"};
+static const std::vector<std::string>& GetValuesAsString() {
+  static const std::vector<std::string>* values =
+      new std::vector<std::string>{"0",
+                                   "1.23",
+                                   "12.345e6",
+                                   "-12.345e-6",
+                                   "123456789.123456789",
+                                   "1231234567890.451234567890"};
+  return *values;
+}
+
+static std::vector<std::pair<Decimal128, int32_t>> BuildDecimalValuesAndScales() {
```

Comment:
nit: making a struct instead of pair can make the code easier to read.

Reply:
Done.